### DISCUSSION TITLE
Using field conditions to validate data coming in through REST API/GraphQL

### DISCUSSION THREAD

**Discussion by tmscer** [directus/directus#11479] [Archive]
I've been trying to configure my collection that has a `title` field which I want to validate to be not empty. This is the field configuration I'm using (and I also have the field set as required):

```json
{
    "conditions": [
        {
            "name": "not-empty",
            "rule": {
                "_and": [
                    {
                        "title": {
                            "_nempty": true
                        }
                    }
                ]
            }
        }
    ]
}
```

Directus doesn't allow my to save an entity without `title` in Directus' UI, but when I try to update `title` to an empty string (`""`) through a GraphQL mutation, the entity is saved without errors. I can, of course, write a filter hook to solve this. But this seems to be like I don't know what to configure or a bug.

Thanks for any suggestions!

**Comment by azrikahar** [comment]
Hi! Just to prevent any misunderstanding from my part, are you configuring said field conditions via the following tab on a particular field? Likely it is since you have mentioned "I have the field set as required":

![image](https://user-images.githubusercontent.com/42867097/152911673-a2e78884-23cf-48b7-a7c1-7422cf591468.png)

If it is, then that is more so "conditions" based on other fields, such as hiding an input when a toggle is true, making a field required when they selected a specific option in a dropdown etc. You can read more here: https://docs.directus.io/configuration/data-model/#conditions

as for what you are trying to achieve, you should use Permissions, particularly Field Validation to ensure this rule is applies to REST/GraphQL in general for a particular role. Here's a quick example of adding the not empty rule for Update operations on "name" field for "test" collection to Public role:

https://user-images.githubusercontent.com/42867097/152913381-c54d6757-3d78-47e1-bc20-706b5b084736.mp4

Link to docs for further reference: https://docs.directus.io/configuration/users-roles-permissions/#configuring-permissions

**Reply by tmscer** [reply]
Thanks a lot for the effort and the video!

> Hi! Just to prevent any misunderstanding from my part, are you configuring said field conditions via the following tab on a particular field? Likely it is since you have mentioned "I have the field set as required":

By that I meant that the field is set as required in the "Field" tab when editing field's configuration.

Does your comment mean that if I want to have some validation rules that apply to all roles (such as a `title` field being not empty), I have to copy that validation configuration across all roles?

**Reply by azrikahar** [reply]
> By that I meant that the field is set as required in the "Field" tab when editing field's configuration.

Oh right, I just recalled that this was logged as an issue over here: https://github.com/directus/directus/issues/11057, which should be what you are facing as you did specify it to be happening (or lack there of) when you are updating.

> Does your comment mean that if I want to have some validation rules that apply to all roles (such as a title field being not empty), I have to copy that validation configuration across all roles?

Yes that is correct, at least for now. I do also recall this being in a discussion somewhere, but the current understandably not so ideal workaround would be to click on the "Rule" field, you should be able to "Copy Raw Value" and subsequently paste it to other roles which should at least make it somewhat easier compared to manually configuring one by one.

Edit: Of course you can also use the permissions API to automate this as well!
